### PR TITLE
Adding capability of running automated process rules tests

### PR DESCRIPTION
**Pull request by jeniawhite**
Enhancing the tests framework to automatically evaluate process rules

Fixes https://github.com/elastic/cloudbeat/issues/157

### CODE REVIEW

File: tests/commonlib/io_utils.py [hunk 1]
```diff
@@ -69,14 +70,53 @@ def exec_command(container_name: str, command: str, param_value: str, resource:
...
+            command = [x for x in command if uskey != x.split("=")[0]]
+
+        r_file["spec"]["containers"][0]["command"] = command
+
+        with current_resource.open(mode="w") as f:
+            yaml.dump(r_file, f)
+
+        # Wait for process reboot
+        # TODO: Implement a more optimal way of waiting
+        time.sleep(60)
```
Comment (gurevichdmitry) [inline tests/commonlib/io_utils.py:122]: Not a good idea to put sleep function on infra level.

File: tests/product/tests/test_rules.py [hunk 2]
```diff
@@ -109,7 +114,936 @@ def check_logs(k8s, timeout, pod_name, namespace, rule_tag, expected) -> bool:
...
+        #         "unset": [
+        #             "--etcd-cafile"
+        #         ]
+        #     },
+        #     '/etc/kubernetes/manifests/kube-apiserver.yaml',
+        #     'failed'
+        # ),
+    ],
+)
+def test_process_rules(data,
```
Comment (gurevichdmitry) [inline tests/product/tests/test_rules.py:1018]: It is possible to create a separate file for process rule tests.

**Review by gurevichdmitry — CHANGES_REQUESTED** [review]
Please create a separate file for process rule tests and check if it possible to move sleep method to test level

**Review by gurevichdmitry — APPROVED** [review]
LGTM

**Review by DaveSys911 — APPROVED** [review]
LGTM

File: tests/commonlib/io_utils.py [hunk 3]
```diff
@@ -100,14 +100,79 @@ def exec_command(container_name: str, command: str, param_value: str, resource:
...
-                raise Exception("User and group parameter shall be separated by ':' ")
+                raise Exception(
+                    "User and group parameter shall be separated by ':' ")
             shutil.chown(path=resource, user=uid_gid[0], group=uid_gid[1])
         else:
-            raise Exception(f"Command '{command}' still not implemented in test framework")
+            raise Exception(
+                f"Command '{command}' still not implemented in test framework")
+
+    @staticmethod
```
Comment (uri-weisman) [inline tests/commonlib/io_utils.py:118]: Can we plz add some description to the function?

File: tests/commonlib/io_utils.py [hunk 4]
```diff
@@ -100,14 +100,79 @@ def exec_command(container_name: str, command: str, param_value: str, resource:
...
+                f"File {resource} does not exist or mount missing.")
+
+        with current_resource.open() as f:
+            r_file = yaml.safe_load(f)
+
+        command = r_file["spec"]["containers"][0]["command"]
+        set_dict = dictionary.get("set", {})
+        unset_list = dictionary.get("unset", [])
+
+        for skey, svalue in set_dict.items():
```
Comment (uri-weisman) [inline tests/commonlib/io_utils.py:135]: Not very easy to understand from the first glimpse, an explanation or even simplifying code will be great.

File: tests/product/tests/test_process_api_server_rules.py [hunk 5]
```diff
@@ -0,0 +1,62 @@
+"""
```
Comment (uri-weisman) [inline tests/product/tests/test_process_api_server_rules.py:1]: Is there a reason to create a test file for each process, from first sight logic looks quite similar?

**Review by uri-weisman — CHANGES_REQUESTED** [review]
GJ Jenia!! 💪 
Added few comments...

**Review by uri-weisman — APPROVED** [review]

**Pull request merged**